tests/components/unifi/test_switch.py

```diff
@@ -1072,6 +1103,74 @@ async def test_dpi_switches_add_second_app(
     assert hass.states.get("switch.block_media_streaming").state == STATE_ON
 
 
+@pytest.mark.parametrize(
+    ("traffic_rule_payload"),
+    [
+        ([TRAFFIC_RULE]),
+    ],
+)
+@pytest.mark.usefixtures("config_entry_setup")
+async def test_traffic_rules(
+    hass: HomeAssistant,
+    aioclient_mock: AiohttpClientMocker,
+    mock_websocket_message,
+    config_entry_setup: ConfigEntry,
+    traffic_rule_payload: list[dict[str, Any]],
+) -> None:
+    """Test control of UniFi traffic rules."""
+
+    assert len(hass.states.async_entity_ids(SWITCH_DOMAIN)) == 1
+
+    # Validate state object
+    switch_1 = hass.states.get("switch.unifi_network_test_traffic_rule")
+    assert switch_1 is not None
+    assert switch_1.state == STATE_ON
+    assert switch_1.attributes.get(ATTR_DEVICE_CLASS) == SwitchDeviceClass.SWITCH
+
+    # Update state object
+
+    device_1 = deepcopy(traffic_rule_payload[0])
+    mock_websocket_message(message=MessageKey.DEVICE, data=device_1)
+    await hass.async_block_till_done()
```

Comment:
I don't understand this one, there are no web sockets related to traffic rules, is this just stale code doing nothing?

Reply:
Just a copy/paste. My first time in this codebase (and I'm not a python dev IRL)

Reply:
> Just a copy/paste. My first time in this codebase (and I'm not a python dev IRL)


![](https://media.tenor.com/oHmGCfehERUAAAAe/yoda-youwillbe.png)